### ISSUE TITLE
freeze after login

### ISSUE THREAD

**Issue by tuananh** [pop-os/cosmic-comp#381]
after install cosmic (both way via apt or building from src (just sysext)), i can only login the first time.

after logging out and logging in again, the desktop freeze

```
Mar 25 22:55:24 twin-engines cosmic-session[70150]: [2024-03-25T15:55:24Z ERROR cosmic_workspaces] Failed to load compositor config: [NoConfigDirectory, NoConfigDirectory, NoConfigDirectory, NoConfigDirectory, NoConfigDirectory, NoConfigDirectory, NoConfigDirectory, NoConfigDirectory]
Mar 25 22:55:24 twin-engines cosmic-session[70150]: [2024-03-25T15:55:24Z ERROR cosmic_workspaces] Failed to load workspaces config: [NoConfigDirectory, NoConfigDirectory]
Mar 25 22:55:24 twin-engines /usr/libexec/gdm-wayland-session[70150]: 2024-03-25T15:55:24.821690Z  WARN cosmic-workspaces: cosmic_session: [2024-03-25T15:55:24Z ERROR cosmic_workspaces] Failed to load compositor config: [NoConfigDirectory, NoConfigDirectory, NoConfigDirectory, NoConfigDirect>
Mar 25 22:55:24 twin-engines /usr/libexec/gdm-wayland-session[70150]: 2024-03-25T15:55:24.821715Z  WARN cosmic-workspaces: cosmic_session: [2024-03-25T15:55:24Z ERROR cosmic_workspaces] Failed to load workspaces config: [NoConfigDirectory, NoConfigDirectory]
Mar 25 22:55:24 twin-engines cosmic-session[70150]: [2024-03-25T15:55:24Z ERROR cosmic_workspaces] Failed to load workspaces config: [NoConfigDirectory, NoConfigDirectory]
Mar 25 22:55:24 twin-engines /usr/libexec/gdm-wayland-session[70150]: 2024-03-25T15:55:24.825830Z  WARN cosmic-workspaces: cosmic_session: [2024-03-25T15:55:24Z ERROR cosmic_workspaces] Failed to load workspaces config: [NoConfigDirectory, NoConfigDirectory]
Mar 25 22:55:24 twin-engines cosmic-session[70150]: [2024-03-25T15:55:24Z ERROR cosmic_workspaces] Failed to load compositor config: [NoConfigDirectory, NoConfigDirectory, NoConfigDirectory, NoConfigDirectory, NoConfigDirectory, NoConfigDirectory, NoConfigDirectory, NoConfigDirectory]
Mar 25 22:55:24 twin-engines /usr/libexec/gdm-wayland-session[70150]: 2024-03-25T15:55:24.826098Z  WARN cosmic-workspaces: cosmic_session: [2024-03-25T15:55:24Z ERROR cosmic_workspaces] Failed to load compositor config: [NoConfigDirectory, NoConfigDirectory, NoConfigDirectory, NoConfigDirect>
Mar 25 22:55:26 twin-engines systemd[1]: systemd-localed.service: Deactivated successfully.
```

originally filed in workspace repo but it seems not related and here i am

my setup
- 22.04
- dual nvidia 3090

**Comment by ids1024** [comment]
What specific driver version? The 550.67 driver released last week advertises a fix for a hang with wgpu (with could at least affect cosmic apps; not sure exactly what was hanging).

Two Nvidia cards of the same model is an interesting configuration that I'm not sure cosmic-comp has been tested with yet. Though I can't specifically think of ways it would cause problems.

**Comment by tuananh** [comment]
the exact version is 550.54.14

**Comment by tuananh** [comment]
@ids1024 is there a flag i can turn on for more verbose logs?

**Comment by mmstick** [comment]
@tuananh Try with the NVIDIA 550.67 driver. Run `cosmic-session` from a TTY shell.

**Comment by tuananh** [comment]
> > the exact version is 550.54.14
> > it's so slow to the point i move the cursor and 30 seconds later, the cursor moves a little bit.
> 
> Your problem is very similar to mine, but it fixed by nvidia 550.54.14 proprietary driver for me
> 
> maybe you can do some of the things what I did before env add `WGPU_BACKEND=vulkan` add kernel command line : `nvidia_drm.modeset=1` and `nvidia_drm.fbdev=1`

can you be more specific? where do i add that env ?

**Comment by skygrango** [comment]
> > > the exact version is 550.54.14
> > > it's so slow to the point i move the cursor and 30 seconds later, the cursor moves a little bit.
> > 
> > 
> > Your problem is very similar to mine, but it fixed by nvidia 550.54.14 proprietary driver for me
> > maybe you can do some of the things what I did before env add `WGPU_BACKEND=vulkan` add kernel command line : `nvidia_drm.modeset=1` and `nvidia_drm.fbdev=1`
> 
> can you be more specific? where do i add that env ?

my current driver verion : 550.67-2

here is my env in `/etc/environment`

```
WGPU_BACKEND=vulkan
GBM_BACKEND=nvidia-drm
```
check linux boot command line have `nvidia_drm.modeset=1` and `nvidia_drm.fbdev=1`

if you have integrated GPU, you can try to disable it in BIOS.

I hope these help you.

**Comment by tuananh** [comment]
Thanks @skygrango but it doesn't seem to fix it for me.

**Comment by skygrango** [comment]
> Thanks @skygrango but it doesn't seem to fix it for me.

I'm sorry to hear this
I use Archlinux, maybe there are some newer packages that allow nvidia to work more properly
recently linux-drm-syncobj-v1 was released, maybe we can wait for cosmic to support it

**Comment by Drakulix** [comment]
The previous discussion here is not very productive, as I am not sure we even have established a good description of the problem.

From what I have gathered "the desktop freezes" means it is starting up properly? With panel/background/dock/etc? But the mouse essentially won't move? Does it react to shortcuts?

> recently linux-drm-syncobj-v1 was released, maybe we can wait for cosmic to support it

That has nothing to do with these symptoms. Just because the system is nvidia based, there is no reason to expect any type of buggy behaviour. The only issue this driver *should* cause is flickering.

**Comment by tuananh** [comment]
> From what I have gathered \"the desktop freezes\" means it is starting up properly? With panel/background/dock/etc? But the mouse essentially won't move? Does it react to shortcuts?

Yes. It loaded but mouse is almost frozen, like i move it and 30-60s later, it will move

**Comment by Drakulix** [comment]
> > From what I have gathered "the desktop freezes" means it is starting up properly? With panel/background/dock/etc? But the mouse essentially won't move? Does it react to shortcuts?
> 
> Yes. It loaded but mouse is almost frozen, like i move it and 30-60s later, it will move

Does it react faster to shortcuts? E.g. to open the launcher?

**Comment by skygrango** [comment]
> > > From what I have gathered "the desktop freezes" means it is starting up properly? With panel/background/dock/etc? But the mouse essentially won't move? Does it react to shortcuts?
> > 
> > 
> > Yes. It loaded but mouse is almost frozen, like i move it and 30-60s later, it will move
> 
> Does it react faster to shortcuts? E.g. to open the launcher?

you are right, the solutions I mentioned are just guesswork, I just upgraded my system and drivers and suddenly everything worked fine, except for some weird flickering and top bar and dock behavior

If you need information on my current setup, please let me know.

@tuananh Sorry for providing inappropriate information, I just heard that linux-drm-syncobj-v1 can solve the strange or wrong behavior of nvidia drivers on linux wayland platforms. but that's probably not the right message or I misunderstood it.

**Comment by tuananh** [comment]
> > > From what I have gathered "the desktop freezes" means it is starting up properly? With panel/background/dock/etc? But the mouse essentially won't move? Does it react to shortcuts?
> > 
> > 
> > Yes. It loaded but mouse is almost frozen, like i move it and 30-60s later, it will move
> 
> Does it react faster to shortcuts? E.g. to open the launcher?

i haven't tried again yet but the menubar appear quite slow as well so i'm guessing shortcut would be similiar.

**Comment by LiamillionSS** [comment]
This issue hasnt been updated since the alpha release, looks like all points concerns raised have been addressed

**Comment by leviport** [comment]
It is pretty old. Closing for now. It can be reopened later if needed.
